hilti/toolchain/src/compiler/resolver.cc

```diff
@@ -66,7 +66,10 @@ struct VisitorPass1 : visitor::MutatingPostOrder {
         }
 
         if ( n->resolvedTypeIndex() ) {
-            if ( auto resolved = n->resolvedType(); resolved->isOnHeap() ) {
+            auto resolved = n->resolvedType();
+            if ( ! resolved )
+                n->addError(util::fmt("type %s cannot be resolved by its name", n->id()));
```

Comment:
Nit: let's put the type into quotes,

```suggestion
                n->addError(util::fmt("type '%s' cannot be resolved by its name", n->id()));
```